### PR TITLE
Models galore

### PR DESCRIPTION
**Pull request by navneeth**
Hello, 
The current example for guided backpropogation only supports Alexnet which is a quite old. I've added an example to allow loading any supported model from the pytorch zoo and visualizing the output of guided backpropagation on it. Please review and merge it.
Specific enhancements:
- The visualize method assumes normalization parameters of imagenet & doesn't support changing this. Edited the class interface to allow these to be specified. However it defaults to imagenet values so it does not affect past behavior.
- The examples queries the tochvision.models module for supported models and allows user to visualize backdrop output on the pre-trained models from pytorch repo.

enjoy,
nav

### CODE REVIEW

**Comment by navneeth** [comment]
Hello @MisaOgura - Do check out and comment on this PR. 
It adds the ability to load and visualize guided backprop output on any network in the pytorch zoo.

**Review by MisaOgura — CHANGES_REQUESTED** [review]
Hi @navneeth, thanks for this PR and apologies for the late reply - hope you are well and staying safe.

Generally speaking, only changes that affects the core functionality of `FlashTorch` are approved as explained [here](https://github.com/MisaOgura/flashtorch/blob/master/CONTRIBUTING.md).

I have also set up automated linting & tests in CI.

Therefore, if you could:

1. Remove `examples/visualize_saliency_with_backprop.py` from the PR
2. Pull the latest changes from the master, merge to your branch and push again so the tests run on the PR

Once that's done and the build passes, I'm happy for this to be merged.

Thanks!

**Comment by navneeth** [comment]
Hi @MisaOgura ,
I did both steps 1 & 2 you mention above i.e.
>1.    Remove examples/visualize_saliency_with_backprop.py from the PR
>2.   Pull the latest changes from the master, merge to your branch and push again so the tests run on the PR

Please check this commit. https://github.com/MisaOgura/flashtorch/compare/master...navneeth:master 
Do I need to submit a new PR or this suffices ?
